### PR TITLE
Add MSA support for all of life's molecules

### PR DESCRIPTION
**Pull request by amorehead**
* Adds MSA support for all of life's molecules
* Tl;dr is, each MSA file corresponds to one specific polymer chain, and (by way of the clustering script's design) each chain's MSA is of a single molecule type even if the chain contains e.g., some DNA residues mixed with the protein chain's amino acids. In this mixed molecule type scenario, this new MSA parsing code treats these "minority-type" residues in each chain as the corresponding unknown residue type of that molecule's type (e.g., DNA residues in a protein-majority chain are treated as `DN` residues in the final MSA, to give the model some partial additional context about these mixed type chains)
* Ligands and modified polymer residues are simply treated as unknown amino acids (with `restype=20`) in the final MSAs
* **TODO:** For MSA chain pairing, an analogous pattern to `_UNIPROT_PATTERN` in `msa_parsing.py` would need to be defined for RNA & DNA entities (@milot-mirdita)

### CODE REVIEW

**Comment by amorehead** [comment]
Whew, got the tests fixed

**Comment by lucidrains** [comment]
@amorehead  i need to read this Deep Work book you told me about
